### ISSUE TITLE
Support for Southern Altai

### ISSUE THREAD

**Issue by jbezos** [latex3/babel#88]
@Matthew-Tate-Scarbrough 

Matthew,

From https://github.com/reutenauer/polyglossia/issues/432#issuecomment-664157332

> I don't intend on writing a babel package if I don't have to. 

You don't have to. Just send to me the requirements and the necessary data and I can create either an `ldf` file or an `ini`/`tex` locale for you. For the latter, there is even a template here:

https://github.com/latex3/babel/tree/e778ff71748688ed790416be21b45f67e065cf1e/locale-templates/alt

**Comment by Matthew-Tate-Scarbrough** [comment]
What all do you need? (beyond the names for the various parts [chapter, index, glossary, etc.])

I only have the names of prefaces, chapters, figures, ToC's, and glossaries right now, which are all: Сӧс бажы; Бажалык; Јурук; Бажалыктар; and Сӧзлик, in the order that I have listed them (all names singular, of course.) So, _Chapter 1_ is _Бажалык 1_.

However, I am gonna try and contact a Library in the Altai Republic and see if I can get a list. IDK how that will go—I can't speak Russian, so... I suppose I could always get a Russian to do it. But, I know the names of those things in Russian too, so, it isn't hard to ask for a list.

**Comment by jbezos** [comment]
@Matthew-Tate-Scarbrough 

• caption names, but note the style will work even without them, because `babel` doesn't raise an error (instead, the user will see a warning with instructions about how to supply the string);
• date format, with the month names, if possible;
• values for righthyphenmin (the minimal number of letters at the end of a line when hyphenated, typically 2 or 3), lefthyphenmin (same at the start of a line, typically 2), the name of the hyphenation patterns (I've read there aren't yet; no problem), and frenchspacing (typically `yes`);
• also useful is an ‘alphabetic’ numeral to be used in `enumerate`, like `а б в г д е ж з` etc.
• specific typographical rules, if there are and you know them.

None of these are strictly necessary, but the more the better.

**Comment by Matthew-Tate-Scarbrough** [comment]
In that case, I will give it my best hand—I do like using `pdflatex` a lot, so. I realised that the `polyglossia` package wasn't very hard, and it seems, based on the two templates that were graciously made, the `babel` one shouldn't be too much harder.

**Comment by jbezos** [comment]
I'll prepare an `ldf` file as well a more complete `tex`/`ini`pair, which I'll add to my personal repositories, so that you can fork it. As to the letters Јј, Ҥҥ, they are available in the T2A and T2B encodings.

**Comment by Matthew-Tate-Scarbrough** [comment]
> I'll prepare an `ldf` file as well a more complete `tex`/`ini`pair

Like an LDF for `polyglossia`? or is that just a generic name, like "language data file?" IDK. lol. Whatever you do, and I thank you, it would be best to base it as closely to Russian as possible, as Altai basically uses all the Russian standards. Of course, I wrote my LDF for polyglossia, I just need to finish getting the other document part names, which I am having a bit of trouble getting, sadly.

**Comment by jbezos** [comment]
`ldf` is ‘language definition file’, which `polyglossia` took for its files (I think for consistency). Allow me a few days to sort things out. The Russian style (by Igor A. Kotelnikov) has lots of bells and whistles, so very likely I'll make a selection of the most relevant features.

**Comment by jbezos** [comment]
@Matthew-Tate-Scarbrough 

Here is a preliminary version: https://github.com/jbezos/southern-altai

A test file (the text is actually Russian):

```
\documentclass{article}

\usepackage[english,southernaltai]{babel}

\ifx\directlua\undefined
  \usepackage[T1,T2A]{fontenc}
  \babelensure[encoding=T2A]{southernaltai}
\else
  \babelfont{rm}{CMU Serif}
\fi

\begin{document}

\selectlanguage{english}

Latin--áçêßøìü.

\selectlanguage{southernaltai}

Јј, Ӧӧ, Ҥҥ, Ӱӱ

В дореволюционной литературе северные алтайцы (тубалары) были известны
как черневые татары.

\selectlanguage{english}

Latin--áçêßøìü \foreignlanguage{southernaltai}{северными}
Latin--áçêßøìü.

\selectlanguage{southernaltai}

В дореволюционной литературе северные
\foreignlanguage{english}{Latin--çêßøìü} алтайцы (тубалары) были
известны как черневые татары.

\end{document}
```

**Comment by Matthew-Tate-Scarbrough** [comment]
Good; I managed to get in contact with an Altai speaker to translate the terminology; I believe I found a few research papers in her name, so she should have a very good idea of the terms and be able to translate them. Once I can get that, I will get also confirmation on how the date should be written, if I cannot find anything, and then fork your repo for the files, then open a pull request. But, if you can point me to a written source or an example in the babel code, how would I write a hyphenation file for it?

Now, also, I would like to ask about the Latin Script. For the polyglossia `.ldf`, they suggested me to have the language as _Altai_, make the variant _southern_, and the script _cyrillic_.

**Comment by jbezos** [comment]
In `babel` there aren't variants on purpose, for practical reasons in the way languages are handled by `babel`, and also for conceptual ones, which in this case is: are Southern Altai and Northern Altai actually variants of the same language or two different languages with similar names? Things like that happen, and there isn't often a clear answer. So I'm following the current customary approach in most systems: each locale is, well, a separate locale.

According to the Unicode CLDR, the `likely` tag for Southern Altai is `alt_Cyrl_RU`, so I presume the default `southernaltai` should be the Cyrillic one. There can be, of course, another `ldf` with a slightly different name (just let me know which name makes sense for you). If you think these names are too long, there is always the possibility to create a shorter one.

As to the hyphenation patterns, it's not a trivial task, and `patgen` with its arcane parameters is severely underdocumented, but things are a lot simpler if they are based on orthographical rules (not morphological), of the kind ‘divide between vowel and consonant’. The patterns are the same for `babel`, `polyglossia` or any other package.

**Comment by Matthew-Tate-Scarbrough** [comment]
Okay, I don't think that the Latin alphabet is that important, as it was only used for 12 years and extent to very, very few works.
But, I do think it could benefit by giving it an alias of, `altai`, over `southernaltai`. The two main dialect families are `Southern` and `Northern`. The Southern ones are very similar, where the Northern ones are very conservative, if I recall. But under the USSR, they chose a specific dialect of the Southern family, called Southern Altai, to be the Standard language, as it was very uniform, IIRC. And Standard Altai, which is Southern Altai, is even more regular than the original Southern Altai they chose. All this to say, “Altai,” is a synonymous name for “Altai Proper.”

The likelihood of anyone needing to typeset any of the other languages is extremely slim, and I am sure I am the only person to ever want to do this in large scale. So in the future, I imagine any dialects or language, political or actual, could just be given other names beside `altai`: `naltai` (Northern Altai), `telengeutaltai`, etc. I understand and respect your decision to just not worry about if it is a different language or not—is Scotts a dialect or language? (IMHO, it is clearly a language just highly mutually intelligible, otherwise Norwegian, Swedish, and Danish are all dialects.)

I will look through the `babel` source and see if I can find any examples of hyphenations files. Now I know what `patgen` is, I can look into that a bit. Turkic languages are agglutinative and tend to be very regular. I have noticed that in speech, there are clear pauses, in Altai, between the base word and its affixes and prefixes, such that make it sound as what I would consider a separate word, in English, even if it can mean nothing on its own.

**Comment by jbezos** [comment]
I'm going to take a break because I need some time for other tasks. In a couple of weeks I'll resume it.

**Comment by Matthew-Tate-Scarbrough** [comment]
That is good—unfortunately, the Altai Speaker I was able to talk to was unsure of the proper translations, so that is delayed on my end anyway.

**Comment by jbezos** [comment]
Have you managed to find further tranlations? If not, no problem, we can define the remaining captions with `\bbl@nocaptions{text}{macroname}`, which prints ‘text’ and shows a warning inviting the user to define ‘macroname’. 

As to the ldf file names, there are already things like `usorbian` for uppersorbian, so `naltai` and `saltai` should be fine.

**Comment by Matthew-Tate-Scarbrough** [comment]
I wasn't, sadly, able to find further translations.  But I was able to confirm some definitions I had found in a dictionary.  The speaker I spoke with, sadly, did not know the proper definitions for those terms.  I am about to do a pull request.

I got the date working for Altai—I looked at your `babel-spanish` repository, but was unable to understand exactly how your Spanish dates were working.  I am not the best with Tex, I simply don't understand.  So, I just looked the American variant of the `\today` macro with `\show` and cloned it for working with Southern Altai.

I haven't really touched most of it in about a couple weeks or so.  But, I am going to be picking up Southern Altai again in a week or so. I like to work on something for a few months, and break for a moth, and pick up again later.  It is possible I _may_ find more technical vocabulary in the future.  But that can always be added in later.
